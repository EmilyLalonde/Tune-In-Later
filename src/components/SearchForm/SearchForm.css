.search-bar {
  border: 3px solid #002236;
  border-radius: 6px;
  display: flex;
  margin-left: 30%;
  margin-top: 20px;
  margin-bottom: 30px;
  height: 40px;
  width: 40%;
}

.search {
  border: none;
  font-size: 20px;
  height: 38px;
  width: 80%
}

.search-button {
  border: none;
  background-color: #36abdd;
  color: #002236;
  font-size: 26px;
  height: 40px;
  width: 20%;
}