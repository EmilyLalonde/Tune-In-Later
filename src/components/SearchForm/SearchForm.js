import React, { Component } from "react";
import {getArtistAlbums} from '../../apiCalls/apiCalls';
import './SearchForm.css';
import WelcomeContainer from '../WelcomeContainer/WelcomeContainer'

class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
        artist: '',
        albums: [],
        error: ''
    }
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value})
  }

  async handleSubmit () {
    this.setState({albums: []})
    this.setState({artist:''})
    try {
        const albums = await getArtistAlbums(this.state.artist);
        this.setState({ albums });
      } catch ({ message }) {
        this.setState({ error: message });
      }
  }

  render() {
    return (
      <section>
        <div className="search-bar">
          <input type="text" placeholder="Search for an Artist..." name="artist" className='search' value={this.state.artist} onChange={this.handleChange}/>
          <button className="search-button" onClick={this.handleSubmit}>Submit</button>
        </div>
        {!!this.state.albums.length && <h2 className="artist-h2">{this.state.artist}</h2>}
        {!!this.state.albums.length && <WelcomeContainer albums={this.state.albums} handleFavorite={this.props.handleFavorite}/>}
      </section>
    );
  }
}

export default SearchForm;