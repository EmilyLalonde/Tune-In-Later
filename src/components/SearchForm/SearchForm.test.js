import React from 'react';
import { shallow } from 'enzyme'
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchForm />)
  });

  it('should match the snapshot with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});