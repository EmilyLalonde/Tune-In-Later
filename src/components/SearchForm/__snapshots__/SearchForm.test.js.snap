// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with the correct props 1`] = `
<div
  className="search-bar"
>
  <input
    className="search"
    name="artist"
    onChange={[Function]}
    placeholder="Search for an Artist"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
